src/dune_engine/execution_parameters.ml

```diff
@@ -0,0 +1,20 @@
+open Stdune
+
+type t = { dune_version : Dune_lang.Syntax.Version.t }
+
+let make ~dune_version = { dune_version }
+
+let equal { dune_version } t =
+  Dune_lang.Syntax.Version.equal dune_version t.dune_version
+
+let hash = Hashtbl.hash
```

Comment:
I think `hash` and `equal` should either both be polymorphic, or both manually written because it's important for them to be consistent.

My vote is:
```
let hash { dune_version } = Dune_lang.Syntax.Version.hash dune_version
```

Reply:
Sounds good, I took your hash function